internal/metrics/info.go

```diff
@@ -0,0 +1,150 @@
+	buildInfo      *metric.Int64Gauge
+	policyCount    *metric.Int64DerivedGauge
+	configChecksum *metric.Int64Gauge
+	metricRegistry = metric.NewRegistry()
+
+	// ConfigLastReloadView contains the timestamp the configuration was last reloaded, labeled by service
+	ConfigLastReloadView = &view.View{
+		Name:        configLastReload.Name(),
+		Description: configLastReload.Description(),
+		Measure:     configLastReload,
+		TagKeys:     []tag.Key{keyService},
+		Aggregation: view.LastValue(),
+	}
+
+	// ConfigLastReloadSuccessView contains the result of the last configuration reload, labeled by service
+	ConfigLastReloadSuccessView = &view.View{
+		Name:        configLastReloadSuccess.Name(),
+		Description: configLastReloadSuccess.Description(),
+		Measure:     configLastReloadSuccess,
+		TagKeys:     []tag.Key{keyService},
+		Aggregation: view.LastValue(),
+	}
+)
+
+// SetBuildInfo records the pomerium build info.  You must call RegisterInfoMetrics to
+// have this exported
+func SetBuildInfo(service string) {
+	if buildInfo == nil {
+		buildInfoL, err := metricRegistry.AddInt64Gauge("build_info",
+			metric.WithDescription("Build Metadata"),
+			metric.WithLabelKeys("service", "version", "revision", "goversion"),
+		)
+		if err != nil {
+			log.Error().Err(err).Msg("internal/metrics: failed to register build info metric")
+			return
+		}
+		buildInfo = buildInfoL
+	}
+
+	m, err := buildInfo.GetEntry(
+		metricdata.NewLabelValue(service),
+		metricdata.NewLabelValue(version.FullVersion()),
+		metricdata.NewLabelValue(version.GitCommit),
+		metricdata.NewLabelValue((runtime.Version())),
+	)
+	if err != nil {
+		log.Error().Err(err).Msg("internal/metrics: failed to add build info metric")
+	}
+	m.Set(1)
+}
+
+// SetConfigInfo records the status, checksum and timestamp of a configuration reload.  You must register InfoViews or the related
+// config views before calling
+func SetConfigInfo(service string, success bool, checksum string) {
+
+	if success {
+		serviceTag := tag.Insert(keyService, service)
+		if err := stats.RecordWithTags(
+			context.Background(),
+			[]tag.Mutator{serviceTag},
+			configLastReload.M(time.Now().Unix()),
+		); err != nil {
+			log.Error().Err(err).Msg("internal/metrics: failed to record config checksum timestamp")
+		}
+
+		if err := stats.RecordWithTags(
+			context.Background(),
+			[]tag.Mutator{serviceTag},
+			configLastReloadSuccess.M(1),
+		); err != nil {
+			log.Error().Err(err).Msg("internal/metrics: failed to record config reload")
+		}
+	} else {
+		stats.Record(context.Background(), configLastReloadSuccess.M(0))
+	}
+}
+
+// Register non-view based metrics registry globally for export
+func RegisterInfoMetrics() {
+	metricproducer.GlobalManager().AddProducer(metricRegistry)
+}
+
+// SetConfigChecksum creates the configuration checksum metric.  You must call RegisterInfoMetrics to
+// have this exported
+func SetConfigChecksum(service string, checksum int64) {
+	if configChecksum == nil {
```

Comment:
This pattern for checking if nil and then initializing the Gauge is repeated.  Open to suggestions on making it more DRY.

Reply:
See above comment about singleton.